build.gradle

```diff
@@ -961,6 +963,53 @@ project(':streams') {
   }
 }
 
+project(':streams:streams-scala') {
+  println "Building project 'streams-scala' with Scala version ${versions.scala}"
+  apply plugin: 'scala'
+  archivesBaseName = "kafka-streams-streams-scala"
+
+  // use experimental flag to enable SAM type conversions in compiler
+  if (versions.baseScala == '2.11') {
+    tasks.withType(ScalaCompile) {
+      scalaCompileOptions.additionalParameters += ["-Xexperimental"]
+    }
+  }
+
+  dependencies {
+    compile project(':streams')
+
+    compile libs.scalaLibrary
+
+    testCompile project(':core')
+    testCompile libs.avro4s
+    testCompile libs.scalaLogging
+
+    testCompile libs.junit
+    testCompile libs.scalatest
+    testCompile libs.scalatestEmbeddedKafka
```

Comment:
This library `scalatestEmbeddedKafka` has a nice integration with scalatest. Hence it makes writing tests easier and we don't have to bother starting / managing the embedded kafka instance. The test code becomes very concise.

Reply:
Generally speaking AK repo tend to avoid dependencies unless it is necessary. I'm wondering if we can improve on Kafka's own EmbeddedKafkaCluster to have the same functionalities as the `net.manub:scalatest-embedded-kafka-streams`.

Reply:
We can definitely use Kafka's own `EmbeddedKafkaCluster` to integrate with ScalaTest. In `net.manub:scalatest-embedded-kafka-streams`, the main value add is integration with ScalaTest and hence you don't have to explicitly start / stop server as part of the test. 

Also with Kafka Streams it has very nice constructs like the one we use here .. https://github.com/lightbend/kafka/blob/scala-streams/streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/StreamToTableJoinScalaIntegrationTestImplicitSerdes.scala#L77-L99 .. Note you just have to define the transformations and do the publish and consume as part of a closure. No need to start / stop the topology. Hence the test code becomes very concise.

Of course it depends on the opinion of the committee but I think this would be a great addition to the dependency.

Here's a suggestion ..

We use `net.manub:scalatest-embedded-kafka-streams` for now. After all it's a *test* dependency. And work on a separate PR to make the integration between `EmbeddedKafkaCluster` and Scalatest better and in line with the functionalities offered by the library.

WDYAT ?

Reply:
I'm a fan of `net.manub:scalatest-embedded-kafka-streams`, but I understand the concern about bringing in more deps.  I only mentioned it on the dev-kafka list because I didn't think there was much value in improving the embedded Kafka implementation in `kafka-stream-scala` and making it a public interface because `scalatest-embedded-kafka` already existed.  I wasn't aware of `EmbeddedKafkaCluster`.

If `scalatest-embedded-kafka` were brought into the project then there will be drift between the version of Kafka broker in code and whatever this test lib references.

@debasishg I like your suggestion:

> work on a separate PR to make the integration between EmbeddedKafkaCluster and Scalatest better and in line with the functionalities offered by the library.

Perhaps we can do this now for this PR, but keep it simple.  I could work on it if you're busy.

Reply:
@seglo @vvcephei @debasishg thanks for your thoughts. We do have plans to publish testing-util artifacts inside AK in the future. And in fact we have been doing so for kafka-streams module as a first step and going to do that for kafka-core and kafka-clients soon. In kafka-clients testing utils we are going to include some improved version of embeddedkafkacluster for users to easily write their integration tests that involve interacting with a mock a kafka cluster.

So I'd suggest we stay with the uglier implementation with the existing embedded kafka cluster and not bring in the dependency.

Reply:
Thanks for your thoughts @guozhangwang .. we will remove the dependency on `net.manub:scalatest-embedded-kafka-streams` and use `EmbeddedKafkaCluster` instead.

Reply:
@guozhangwang - Removed all dependencies on `net.manub:scalatest-embedded-kafka` and `net.manub:scalatest-embedded-kafka-streams`. Now using `EmbeddedKafkaCluster` instead for tests. Also removed the test that used avro - hence dependency on `avros` eliminated.

Reply:
Thanks @debasishg !